Update dependencies

omid opened the pull request:
Update dependencies and a bit of housekeeping.

omid commented:
There is a typo in `Cancelation`, but it's also like this in the app itself, so it's bigger than the sdk and most probably a breaking change!

curquiza commented:
Hello @omid 
thanks for the PR

We are ok with breaking changes since the app is not stable (using version format like v0.X.Y and not vX.Y.Z), so we can accept a PR. The most important is to be clear in the communication

curquiza commented:
Also @omid you said you updated the dependencies, but which ones?

omid commented:
@curquiza Thanks 🙏🏼 

> Also @omid you said you updated the dependencies but you also changed a lot of things

I updated `syn` to 2, `jsonwebtoken` to 9, and removed env_logger, since it wasn't in use.

> I don't feel comfortable accepting a PR where I have that much files to review for different reasons

If you don't mind, the update dep thingy changes are in just one commit which you can find [here](https://github.com/meilisearch/meilisearch-rust/pull/542/commits/61dc8ffc4cf590cb279f87e5eff0dd93aebe81a2).

> We are ok with breaking changes since the app is not stable (using version format like v0.X.Y and not vX.Y.Z), so we can accept a PR. The most important is to be clear in the communication

Sure, the only issue will be that the change should be applied everywhere! Including all SDKs I think, which I'm not familiar with their languages and it's a big chunk of work also.

curquiza commented:
> If you don't mind, the update dep thingy changes are in just one commit which you can find [here](https://github.com/meilisearch/meilisearch-rust/pull/542/commits/61dc8ffc4cf590cb279f87e5eff0dd93aebe81a2).

Not 100% true, this commit also change some parts of the code at the middle of the non-code related update:
https://github.com/meilisearch/meilisearch-rust/pull/542/commits/351651951e50155d4125a6df4ec23b914cfd1e0a

curquiza commented:
>Sure, the only issue will be that the change should be applied everywhere! Including all SDKs I think, which I'm not familiar with their languages and it's a big chunk of work also. (github said [50 files](https://github.com/search?q=org%3Ameilisearch%20cancelation&type=code) in the meilisearch org)

Can you point which kind of change you refer and why the issue is in all the SDKs?

omid commented:
> Can you point which kind of change you refer and why the issue is in all the SDKs?

As I said in the first post here, `There is a typo in *Cancelation*`. It's wrong, it's cancellation, double LL.

omid commented:
> Not 100% true, this commit also change some parts of the code at the middle of the non-code related update:

The update deps are in one commit. Others are typos and some housekeeping. There are some changes in the macro part in other commits, but they are not about the logic. Just code style, typo and so on.

curquiza commented:
Can you fix git conflicts?

omid commented on meilisearch-index-setting-macro/src/lib.rs:
```diff
@@ -123,140 +140,35 @@ fn get_index_config_implementation(
 
-         async fn generate_index(client: &::meilisearch_sdk::client::Client) -> std::result::Result<::meilisearch_sdk::indexes::Index, ::meilisearch_sdk::tasks::Task> {
+         async fn generate_index(client: &::meilisearch_sdk::client::Client) -> Result<::meilisearch_sdk::indexes::Index, ::meilisearch_sdk::tasks::Task> {
             return client.create_index(#index_name, #primary_key_token)
                 .await.unwrap()
-                .wait_for_completion(&client, ::std::option::Option::None, ::std::option::Option::None)
+                .wait_for_completion(&client, None, None)
```
You are absolutely right, my bad.

irevoire commented on meilisearch-index-setting-macro/src/lib.rs:
```diff
@@ -1,10 +1,18 @@
+use structmeta::{Flag, StructMeta};
+use syn::{parse_macro_input, spanned::Spanned};
+
+#[derive(Clone, StructMeta, Default)]
+struct FieldAttrs {
+    primary_key: Flag,
+    displayed: Flag,
+    searchable: Flag,
+    distinct: Flag,
+    filterable: Flag,
+    sortable: Flag,
+}
```
That’s pretty nice, didn’t know about it before

irevoire reviewed (approved):
Hey @omid thanks for your contribution it’s very nice, and updating syn was not an easy task!

We’re very grateful you did it; however, if you do it again, please follow curqui advice.
Do one PR that **only** updates the dependencies and another that only fixes typos.
That will make it way easier for us to review and merge (that’s also why I took so long to get back to you)

bors merge
